compiler/rustc_resolve/src/macros.rs

```diff
@@ -458,6 +460,82 @@ impl<'ra, 'tcx> ResolverExpand for Resolver<'ra, 'tcx> {
         self.path_accessible(expn_id, path, &[MacroNS])
     }
 
+    fn cfg_accessible_crate(&mut self, crate_name: Ident) -> Result<DefId, Indeterminate> {
+        let this = self;
+        let finalize = false;
+        let Some(binding) = this.extern_prelude_get(crate_name, finalize) else {
```

Comment:
One example why time travel is still necessary even with the new "restricted" rules - extern prelude changes as crate expands and `extern_prelude_get` can give different results at different points in time.

Reply:
> `extern_prelude_get` can give different results at different points in time

Now I realize i didn't really understand how the extern prelude works. I'll reach you on Zulip to gain better understanding on this bullet point before i continue.